src/RemoteInboxNotifications/GetRuleProcessor.php

```diff
@@ -59,6 +59,15 @@ public static function get_processor( $rule_type ) {
 				);
 			case 'data':
 				return new DataRuleProcessor();
+			case 'product_count':
+				return new ProductCountRuleProcessor(
+					new \WC_Product_Query(
+						array(
+							'limit'    => 1,
+							'paginate' => 1,
```

Comment:
Do we not want to limit this query to published products?

Also, specifying `'return' => 'ids'` can further reduce the work done in the query class: https://github.com/woocommerce/woocommerce/blob/ee119e0a7ed24cf8c83983ff013bd188e3081d19/includes/data-stores/class-wc-product-data-store-cpt.php#L2004

Nit: `paginate` should be a boolean.